include/dxc/dxcapi.h

```diff
@@ -477,7 +477,9 @@ typedef enum DXC_OUT_KIND {
   DXC_OUT_ROOT_SIGNATURE = 9, // IDxcBlob - Serialized root signature output
   DXC_OUT_EXTRA_OUTPUTS  = 10,// IDxcExtraResults - Extra outputs
   DXC_OUT_REMARKS = 11,       // IDxcBlobUtf8 or IDxcBlobUtf16 - text directed at stdout
+  DXC_OUT_TIME_REPORT = 12,   // IDxcBlobUtf8 or IDxcBlobUtf16 - text directed at stdout
 
+  DXC_OUT_LAST = DXC_OUT_TIME_REPORT, // Last value for a counter
```

Comment:
We should add a `static_assert(DXC_OUT_LAST = DXC_OUT_TIME_REPORT)` so that when we add a new enum value we will get a compile-generated reminder to update it.

Reply:
Unfortunately all that will do is cause us to update in tow places. If we add an enum but don't update `DXC_OUT_LAST`'s assignment the static assert will still pass :(

We could add another field that is auto-incremented to handle that case. If this wasn't part of the dxcapi header I'd like to redo the enum ordering and put the `NONE` value at the end. Then we could have `NONE` be the number of valid cases and simplify the indexing using the enum value. I'm not sure how much code that could break since these are part of our ABI 😢 .

Reply:
Ah yeah of course.

So why not make DXC_OUT_LAST autoincrementing then? We could still have a static assert if we wanted to make it clear that we are relying on that to work correctly

`static_assert(DXC_OUT_LAST == DXC_OUT_TIME_REPORT`, "you added a new enum, which is ok, but update this assert to make sure the count is still correct")`

Reply:
Oh, I just saw your update. That way looks good too.

Reply:
I think DXC_OUT_NONE should be a stable value, so making it last+1 isn't good.

One thing that can be done here is to define DXC_OUT_LAST after DXC_OUT_ENUMS as:
`DXC_OUT_LAST = DXC_OUT_ENUMS - 1`

That way you don't have to manually update the value.